### ISSUE TITLE
bug: not finding HF_TOKEN

### ISSUE THREAD

**Issue by TheKidThatCodes** [bentoml/OpenLLM#1115]
### Describe the bug

tried to download llama 3.2 3b

### To reproduce

standard fresh install in a python 3.13 venv on kubuntu via ssh (shell on both my mac and pc is zsh)
try to download llama
it says login
add my hf token
it still says login
try other methods of adding hf token
still says login

#### Environment variable

```bash
BENTOML_DEBUG=''
BENTOML_QUIET=''
BENTOML_BUNDLE_LOCAL_BUILD=''
BENTOML_DO_NOT_TRACK=''
BENTOML_CONFIG=''
BENTOML_CONFIG_OPTIONS=''
BENTOML_PORT=''
BENTOML_HOST=''
BENTOML_API_WORKERS=''
```

#### System information

`bentoml`: 1.3.14
`python`: 3.13.0
`platform`: Linux-6.11.0-9-generic-x86_64-with-glibc2.40
`uid_gid`: 1000:1000
<details><summary><code>pip_packages</code></summary>

<br>

```
a2wsgi==1.10.7
aiohappyeyeballs==2.4.3
aiohttp==3.11.8
aiosignal==1.3.1
aiosqlite==0.20.0
annotated-types==0.7.0
anyio==4.6.2.post1
appdirs==1.4.4
asgiref==3.8.1
attrs==24.2.0
bentoml==1.3.14
cattrs==23.1.2
certifi==2024.8.30
circus==0.18.0
click==8.1.7
click-option-group==0.5.6
cloudpickle==3.1.0
deepmerge==2.0
deprecated==1.2.15
distro==1.9.0
dulwich==0.22.6
frozenlist==1.5.0
fs==2.4.16
h11==0.14.0
httpcore==1.0.7
httpx==0.27.2
httpx-ws==0.6.2
idna==3.10
importlib-metadata==6.11.0
inflection==0.5.1
jinja2==3.1.4
jiter==0.8.0
markdown-it-py==3.0.0
markupsafe==3.0.2
mdurl==0.1.2
multidict==6.1.0
numpy==2.1.3
nvidia-ml-py==12.560.30
openai==1.52.0
openllm==0.6.14
opentelemetry-api==1.20.0
opentelemetry-instrumentation==0.41b0
opentelemetry-instrumentation-aiohttp-client==0.41b0
opentelemetry-instrumentation-asgi==0.41b0
opentelemetry-sdk==1.20.0
opentelemetry-semantic-conventions==0.41b0
opentelemetry-util-http==0.41b0
packaging==24.2
pathlib==1.0.1
pathspec==0.12.1
pip==24.2
pip-requirements-parser==32.0.1
prometheus-client==0.21.0
prompt-toolkit==3.0.36
propcache==0.2.0
psutil==6.1.0
pyaml==24.9.0
pydantic==2.9.2
pydantic-core==2.23.4
pygments==2.18.0
pyparsing==3.2.0
python-dateutil==2.9.0.post0
python-json-logger==2.0.7
python-multipart==0.0.17
pyyaml==6.0.2
pyzmq==26.2.0
questionary==2.0.1
rich==13.9.4
schema==0.7.7
setuptools==75.6.0
shellingham==1.5.4
simple-di==0.1.5
six==1.16.0
sniffio==1.3.1
starlette==0.41.3
tabulate==0.9.0
tomli-w==1.1.0
tornado==6.4.2
tqdm==4.67.1
typer==0.13.1
typing-extensions==4.12.2
urllib3==2.2.3
uv==0.5.5
uvicorn==0.32.1
watchfiles==1.0.0
wcwidth==0.2.13
wrapt==1.17.0
wsproto==1.2.0
yarl==1.18.0
zipp==3.21.0
```

</details>

### System information (Optional)

128gb ham
rtx 4080 with 16gb vham
amd ryzen 7950x
ubuntu with kubuntu-desktop
this is where u flex ur pc right?
zsh

**Comment by pwyoung** [comment]
I am having the same issue with ubuntu 24.04 in a venv.

**Comment by aarnphm** [comment]
Make sure you have access to the gated repo

**Comment by TheKidThatCodes** [comment]
i have access, openllm is at fault, and i have converted to vLLM, as it seems like a better option for my project (faster)

**Comment by bojiang** [comment]
Hello. This issue is reproduced on my machine, thanks for reporting!

**Comment by bojiang** [comment]
Fixed in 2b9c76f6bfebf4f3bcea3b663ca2280398b39d06
As for using vllm, vllm is also our backend by default for Linux. We also support other backend like llamacpp for MacOS. The performance should be identical between openllm and vllm on Linux.
Thanks for contribition.

**Comment by TheKidThatCodes** [comment]
> Fixed in [2b9c76f](https://github.com/bentoml/OpenLLM/commit/2b9c76f6bfebf4f3bcea3b663ca2280398b39d06) As for using vllm, vllm is the picked default backend on Linux. It is stable and fast. We also support other backend like llamacpp for MacOS. The performance should be identical between openllm and vllm on Linux. Thanks for contribition.

i think i will skip the middleman for this, as vllm seems to do what i need it to do much better
